mi/stream_writer/model.py

```diff
@@ -0,0 +1,171 @@
+import os
+from dataclasses import asdict, dataclass, field, fields
+from datetime import datetime as dt
+from enum import Enum
+from typing import Optional, TypeVar
+
+from pydantic import BaseModel
+
+from mi.stream_writer.constants import PATH_TO_QUERIES, TYPE_SEPARATOR, DateTimeFormats
+from mi.stream_writer.utils import hash_nhs_number, to_snake_case
+
+
+class Environment(BaseModel):
+    POSTGRES_DATABASE_NAME: str
+    RDS_CLUSTER_HOST: str
+    RDS_CLUSTER_PORT: int
+    POSTGRES_USERNAME: str
+    POSTGRES_PASSWORD: str
+
+    @classmethod
+    def construct(cls) -> "Environment":
+        return cls(**os.environ)
+
+
+class SecretsManagerCache:
+    def __init__(self, client):
+        self.client = client
+        self.secret = None
+
+    def get_secret(self, secret_id):
+        if self.secret is None:
+            raw_secret = self.client.get_secret_value(SecretId=secret_id)
+            self.secret = AwsSecret(**raw_secret).SecretString
+        return self.secret
+
+
+@dataclass
+class AwsSecret:
+    ARN: str
+    Name: str
+    VersionId: str
+    SecretString: str
+    SecretBinary: Optional[bytes] = None
+    VersionStages: Optional[list[str]] = None
+    CreatedDate: Optional[dt] = None
+    ResponseMetadata: Optional[dict] = None
+
+
+class DynamodbEventImageType(str, Enum):
+    OLD_IMAGE = "OldImage"
+    NEW_IMAGE = "NewImage"
+
+
+class Action(str, Enum):
+    CREATED = "created"
+    DELETED = "deleted"
+
+
+class DynamoDBEventConfig(BaseModel):
+    """Model to map a DynamoDb Stream Image Type on to a query file"""
+
+    image_type: DynamodbEventImageType
+    action: Action
+
+    @property
+    def sql(self):
+        with open(f"{PATH_TO_QUERIES}/{self.action}.sql") as f:
+            return f.read()
+
+
+class Dimension:
```

Comment:
I understand (I think) whats happening here, but a little run through to understand the benefits would be much appreciated!